src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Measurement.cs

```diff
@@ -57,23 +57,45 @@ public Measurement(T value, params KeyValuePair<string, object?>[]? tags)
         }
 
         /// <summary>
-        /// Initializes a new instance of the Measurement using the value and the list of tags.
+        /// Initializes a new instance of Measurement with the provided <paramref name="value"/> and a <see cref="ReadOnlySpan{T}"/> containing
+        /// zero or more associated <paramref name="tags"/>.
         /// </summary>
-        /// <param name="value">The measurement value.</param>
-        /// <param name="tags">The measurement associated tags list.</param>
+        /// <param name="value">The value of the measurement.</param>
+        /// <param name="tags">The <see cref="KeyValuePair{TKey, TValue}"/> tags associated with the measurement.</param>
         public Measurement(T value, params ReadOnlySpan<KeyValuePair<string, object?>> tags)
         {
             _tags = tags.ToArray();
             Value = value;
         }
 
         /// <summary>
-        /// Gets the measurement tags list.
+        /// Initializes a new instance of the Measurement with the provided <paramref name="value"/> and a <see cref="TagList"/> containing
+        /// zero or more associated <paramref name="tags"/>.
+        /// </summary>
+        /// <param name="value">The value of the measurement.</param>
+        /// <param name="tags">A <see cref="TagList"/> containing the <see cref="KeyValuePair{TKey, TValue}"/> tags associated with the measurement.</param>
+        public Measurement(T value, in TagList tags)
+        {
+            if (tags.Count > 0)
+            {
+                _tags = new KeyValuePair<string, object?>[tags.Count];
+                tags.CopyTo(_tags.AsSpan());
```

Comment:
This will require changing Instrument.RecordMeasurement. https://github.com/dotnet/runtime/blob/66ae90f3b7ec4f13fffcd71913eca0b45777e58c/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.netfx.cs#L77

I am not sure if it is worth doing it?

Reply:
@stephentoub I am going to merge the change. If we need to change this part, we can do it in a separate PR. Are you ok with that?

Reply:
Sure

Reply:
Also, doesn't `Tags` only return from the underlying `_overflowTags`, which may be null if less than 8 items are stored? `CopyTo` includes logic to fallback to ensure the specific tags are copied.

Reply:
Ah, I see that's not the case for the netcore implementation, `Tags` returns in the InlineArray if the overflow is empty. And as @tarekgh then shows, Instrument.netfx handles the fallback.